src/Services/ShipmentNotificationService.php

```diff
@@ -489,8 +489,21 @@ public function prepareRequestDto(Order $order): RequestDTO
             'method' => __METHOD__
           ]
         );
-
-        $scacCode = $this->carrierScacRepository->findScacByCarrierId($plentymarketsShippingInformation->shippingServiceProvider->id);
+        if (empty($plentyMarketsShippingInformation->shippingServiceProvider->id)) {
```

Comment:
if you're checking for "null or empty", you should do an `!isset(...)` check here as well.